### PR TITLE
organize snippets to mirror the directory structure

### PR DESCRIPTION
**Pull request by eshaben**
### Description

Reorganize all of the snippets to match the directory structure, like we do with the images.

### Checklist

- [x] If pages have been moved around, I have run the `move-pages.py` script to move the pages and update the image paths on the chinese repo
    - [x] After the script has been run, I have created an additional PR in `moonbeam-docs-cn`

### Corresponding PRs

Goes with CN PR: https://github.com/moonbeam-foundation/moonbeam-docs-cn/pull/340

### After Translation Requirements

- [x] No additional PRs are required after the translations are done

### CODE REVIEW

File: builders/get-started/explorers.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ Moonbeam provides two different kind of explorers: ones to query the Ethereum AP
```
Comment (themacexpert) [inline builders/get-started/explorers.md:13]: In above line, should be two different kinds of explorers

File: builders/get-started/quick-start.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ When working with developer tools, depending on the tool, you might need to conf
 
 Moonbeam provides two different kind of explorers: ones to query the Ethereum API, and others dedicated to the Substrate API. All EVM-based transactions are accessible via the Ethereum API wheras the Substrate API can be relied upon for Substrate-native functions such as governance, staking, and some information about EVM-based transactions. For more information on each explorer, please check out the [Block Explorers](/builders/get-started/explorers){target=_blank} page.
```
Comment (themacexpert) [inline builders/get-started/quick-start.md:69]:
```suggestion
Moonbeam provides two different kind of explorers: ones to query the Ethereum API, and others dedicated to the Substrate API. All EVM-based transactions are accessible via the Ethereum API whereas the Substrate API can be relied upon for Substrate-native functions such as governance, staking, and some information about EVM-based transactions. For more information on each explorer, please check out the [Block Explorers](/builders/get-started/explorers){target=_blank} page.
```

File: builders/pallets-precompiles/precompiles/eth-mainnet.md [hunk 3]
```diff
@@ -96,7 +96,7 @@ The BN128Mul precompile implements a native elliptic curve multiplication with a
 
 Currently there is no BN128Mul support in Solidity, so it needs to be called with inline assembly. The following sample code can be used to call this precompile.
 
---8<-- 'code/precompiles/bn128mul.md'
+--8<-- 'code/builders/pallets-precompiles/precompiles/eth-mainnet/bn128mul.md'
 
 Using the [Remix compiler and deployment](/builders/build/eth-api/dev-env/remix/){target=_blank} and with [MetaMask pointing to Moonbase Alpha](/tokens/connect/metamask/){target=_blank}, you can deploy the contract and call the `callBn256ScalarMul(bytes32 x, bytes32 y, bytes32 scalar)` method to return the result of the operation.
```
Comment (themacexpert) [inline builders/pallets-precompiles/precompiles/eth-mainnet.md:102]: Below line should say elliptic curve pairing

File: tutorials/interoperability/uniswapv2-swap-xcm.md [hunk 4]
```diff
@@ -34,11 +34,11 @@ Considering all the steps summarized in the [introduction](#introduction), the f
 - You needs to have UNITs on the relay chain to pay for transaction fees when sending the XCM. If you have a Moonbase Alpha account funded with DEV tokens, you can swap some DEV for xcUNIT here on [Moonbeam Swap](https://moonbeam-swap.netlify.app/#/swap){target=_blank}. Then withdraw the xcUNIT from Moonbase Alpha to [your account on the Moonbase relay chain](https://polkadot.js.org/apps/?rpc=wss://frag-moonbase-relay-rpc-ws.g.moonbase.moonbeam.network#/accounts){target=_blank} using [apps.moonbeam.network](https://apps.moonbeam.network/moonbase-alpha/){target=_blank}
```
Comment (themacexpert) [inline tutorials/interoperability/uniswapv2-swap-xcm.md:34]:
```suggestion
- You need to have UNITs on the relay chain to pay for transaction fees when sending the XCM. If you have a Moonbase Alpha account funded with DEV tokens, you can swap some DEV for xcUNIT here on [Moonbeam Swap](https://moonbeam-swap.netlify.app/#/swap){target=_blank}. Then withdraw the xcUNIT from Moonbase Alpha to [your account on the Moonbase relay chain](https://polkadot.js.org/apps/?rpc=wss://frag-moonbase-relay-rpc-ws.g.moonbase.moonbeam.network#/accounts){target=_blank} using [apps.moonbeam.network](https://apps.moonbeam.network/moonbase-alpha/){target=_blank}
```

**Review by themacexpert — APPROVED** [review]
LGTM. I didn't check every single snippet but rather did a "randomly selected audit" and everything checked out :) 

Some random typos I found along the way are noted although they don't have anything to do with this pr so feel free to ignore them or put them in a later pr.

**Comment by eshaben** [comment]
Good catches! Thanks!

**Comment by albertov19** [comment]
This is great @eshaben  - Is there a way to check in the logs when building the site if we missed any?

**Comment by eshaben** [comment]
Not really. I just tried moving a snippet to a different directory to see if there were any errors in the logs, but there weren't 😕

**Comment by themacexpert** [comment]
@eshaben I did a quick check and it looks like the new viem page hasn't been updated yet to the updated disclaimer link, still shows `text/disclaimers/third-party-content.md'`

**Comment by eshaben** [comment]
@themacexpert thanks, yeah I just merged in the latest changes and updated all the other snippets and paths that needed to be updated

**Pull request merged**